### PR TITLE
[RFR][1LP] Small enhancement to appliance.server_roles setter

### PR DESCRIPTION
**Pull request by otsuman**
Added new method to update server_role without providing the whole dict of roles

`appliance.update_server_roles({"automate": True})

### CODE REVIEW

File: utils/appliance/__init__.py [hunk 1]
```diff
@@ -1694,7 +1694,7 @@ def server_roles(self, roles):
         yaml = self.get_yaml_config()
         yaml['server']['role'] = ','.join([role for role, boolean in roles.iteritems() if boolean])
```
Comment (RonnyPfannschmidt) [inline utils/appliance/__init__.py]: based on this line the given description is a lie as the roles arent merged/joined

thecheck is just worse

if the new code works by a hidden interaction, then it deserves a propper code lineup, else this is simply incomprehensible after just a few days

Comment (otsuman) [inline utils/appliance/__init__.py]: done. Just added new method

File: utils/appliance/__init__.py [hunk 2]
```diff
@@ -1696,6 +1696,12 @@ def server_roles(self, roles):
         self.set_yaml_config(yaml)
         wait_for(lambda: self.server_roles == roles, num_sec=300, delay=15)
 
+    def update_server_roles(self, roles):
```
Comment (RonnyPfannschmidt) [inline utils/appliance/__init__.py]: please call that parameter `changed_roles` so its purpose is more apparent

Comment (otsuman) [inline utils/appliance/__init__.py]: done

**Pull request merged**